### DISCUSSION TITLE
Is there any way to use CLI deploy model to kubernetes

### DISCUSSION THREAD

**Discussion by crane136** [bentoml/BentoML#1473] [Q&A]
I want to use bentoml apply command to deplyment model to kubernetes. Also the deployments can be managed by BENTOML UI Tools

	I got some problems When I use the CLI: ( Is there any way to use CLI deploy model to kubernetes?)
		
# iris-classifier

apiVersion: v1
kind: Service
metadata:
    labels:
        app: iris-classifier
    name: iris-classifier
spec:
    ports:
    - name: predict
      port: 5000
      targetPort: 5000
    selector:
      app: iris-classifier
    type: LoadBalancer

---
apiVersion: apps/v1
kind: Deployment
metadata:
    labels:
        app: iris-classifier
    name: iris-classifier
spec:
    selector:
        matchLabels:
            app: iris-classifier
    template:
        metadata:
            labels:
                app: iris-classifier
        spec:
            containers:
            - image: iris-classifier:20210226105119_51D3D8
              imagePullPolicy: IfNotPresent
              name: iris-classifier
              ports:
              - containerPort: 5000

pool-100-8-12-202:models crane$ bentoml deployment apply -f iris-classifier.yaml 
Usage: bentoml deployment apply [OPTIONS]

Error: Invalid value for '-f' / '--file': Input value is not recognizable yaml file or yaml content

# Try another:
apiVersion: serving.kubeflow.org/v1alpha2
kind: InferenceService
metadata:
  labels:
    controller-tools.k8s.io: "1.0"
  name: iris-classifier
spec:
  default:
    predictor:
      custom:
        container:
          image: iris-classifier:20210226105119_51D3D8
          ports:
            - containerPort: 5000

pool-100-8-12-202:models crane$ bentoml deployment apply -f bentoml.yaml
Error: deployment apply failed: Platform "None" is not supported in the current version of BentoML

**Comment by yubozhao** [comment]
@crane136 BentoML doesn't provide Kubernetes deployment at the moment.  This feature is on our road map and we are actively designing and working on it.

I would love to get your input on what you are looking for.  Can you tell me more about your use-cases and current setup?  

I am also curious about the Kubeflow inference service you had in your "Try another" section. Do you use Kubeflow in your workflow? What part of Kubeflow system you guys are using?
